include/__utility.hpp

```diff
@@ -43,7 +43,11 @@ namespace std {
 
   // Some utilities for manipulating lists of types at compile time
   template <class...>
-    struct __types;
+  struct __types
+#if defined(__GNUC__) && !defined(__clang__)
+  {}  // BUGBUG: GCC does not like this "incomplete type"
+#endif
```

Comment:
This is unfortunate. I like having it incomplete because I want to know if I'm accidentally paying for a template instantiation I don't need. I'm ok merging this for now, but I'd like to understand where gcc thinks it needs to instantiate it.

Reply:
Agreed, I haven't minimized this but it was "all over the place".
I suspect that there is some abstraction in the whole instantiation stack that is doing something that GCC does not like, and it just manifested with this.